tools/@aws-cdk/prlint/linter-base.ts

```diff
@@ -0,0 +1,323 @@
+ *
+ * Inheritance is not great, but this was the easiest way to factor this out for now.
+ */
+export class PullRequestLinterBase {
+  /**
+   * Find an open PR for the given commit.
+   * @param sha the commit sha to find the PR of
+   */
+  public static async getPRFromCommit(client: Octokit, owner: string, repo: string, sha: string): Promise<GitHubPr | undefined> {
+    const prs = await client.search.issuesAndPullRequests({
+      q: sha,
+    });
+    console.log('Found PRs: ', prs);
+    const foundPr = prs.data.items.find(pr => pr.state === 'open');
+    if (foundPr) {
+      // need to do this because the list PR response does not have
+      // all the necessary information
+      const pr = (await client.pulls.get({
+        owner,
+        repo,
+        pull_number: foundPr.number,
+      })).data;
+      console.log(`PR: ${foundPr.number}: `, pr);
+      // only process latest commit
+      if (pr.head.sha === sha) {
+        return pr;
+      }
+    }
+    return;
+  }
+
+  protected readonly client: Octokit;
+  protected readonly prParams: { owner: string, repo: string, pull_number: number };
+  protected readonly issueParams: { owner: string, repo: string, issue_number: number };
+  protected readonly linterLogin: string;
+
+  private _pr: GitHubPr | undefined;
+
+  constructor(readonly props: PullRequestLinterBaseProps) {
+    this.client = props.client;
+    this.prParams = { owner: props.owner, repo: props.repo, pull_number: props.number };
+    this.issueParams = { owner: props.owner, repo: props.repo, issue_number: props.number };
+    this.linterLogin = props.linterLogin;
+  }
+
+  public async pr(): Promise<GitHubPr> {
+    if (!this._pr) {
+      const r = await this.client.pulls.get(this.prParams);
+      this._pr = r.data;
+    }
+    return this._pr;
+  }
+
+  /**
+   * Execute the given set of actions
+   */
+  public async executeActions(actions: LinterActions) {
+    const pr = await this.pr();
+
+    for (const label of actions.removeLabels ?? []) {
+      this.removeLabel(label, pr);
+    }
+
+    for (const label of actions.addLabels ?? []) {
+      this.addLabel(label, pr);
+    }
+
+    if (actions.dismissPreviousReview || actions.requestChanges) {
+      if (actions.dismissPreviousReview && actions.requestChanges) {
+        throw new Error(`It does not make sense to supply both dismissPreviousReview and requestChanges: ${JSON.stringify(actions)}`);
+      }
+
+      const existingReview = await this.findExistingPRLinterReview();
+      this.deletePRLinterComment();
+
+      if (actions.dismissPreviousReview) {
+        this.dismissPRLinterReview(existingReview);
+      }
+      if (actions.requestChanges) {
+        this.createOrUpdatePRLinterReview(actions.requestChanges.failures, actions.requestChanges.exemptionRequest, existingReview);
+      }
+    }
+  }
+
+  /**
+   * For code that requires an exception
+   */
+  public actionsToException(actions: LinterActions) {
+    if (actions.requestChanges) {
```

Comment:
Was this always the logic? only fail the linter if it requests changes?

Reply:
Yes.